tmpl/wiki/package-quality.md

```diff
@@ -0,0 +1,138 @@
+## Package Quality
+
+The MirageOS ecosystem of libraries is booming:
+
+```
+$ opam list -a | grep mirage | wc -l
+      90
+```
+
+We now have various libraries covering all aspects of the OS: from network to
+block device drivers -- at all aspects of the stack: from virtual network cards
+to TLS or virtual block devices to virtual filesystems. These libraries also
+span multiple hypervisors including Xen and (via solo5) KVM.
+
+All of this is great: this means that as a MirageOS user, you can spend time
+developing your own application instead of spending time re-implementing
+something that the OS would usually do for you. There are of course still
+gaps in the ecosystem, but things are getting better!
+
+The next problem is knowing the level of confidence to attribute to each of
+these libraries. Would be great to be able to answer questions such as:
+Does the library has a good test coverage? Was it formally verified?
+Was it used in production? In order to do this, I would like to introduce
+the "MirageOS quality levels". Ideally this will live in a tool (e.g.
+`mirage lint`) but to start we list the criteria that package aiming
+for increased quality should follow.
+
+### ![Level 1:](https://img.shields.io/badge/level-1-blue.svg) Follow Packaging Guidelines
+
+See the [packaging guidelines](https://mirage.io/wiki/packaging) for more
+details. In summary:
+
+- The package should work well with [odig](http://erratique.ch/software/odig)
+  e.g. `CHANGES.md`, `LICENSE.md` and `README.md` should exist.
+
+- The package should work well with [topkg](http://erratique.ch/software/topkg),
+  e.g. `pkg/pkg.ml` should exist.
+
+- The use of [jbuilder](http://jbuilder.readthedocs.io/en/latest/) is
+  strongly encouraged but not yet mandatory as 1.0.0 has not been
+  officially released. However, we plan to enforce its use as soon as
+  it is the case -- assuming that it has all the features needed for
+  building MirageOS applications.
```

Comment:
I'm not a big fan of requiring a specific build tool, instead the next point is the important one:  opam and ocamlfind name should be equal!

AFAICT, there are still some open issues with jbuilder -- on the one side the lack of being able to integrate coverage (and profiling!?) information during builds -- on the other side I completely miss any performance comparison (and just recently stumbled upon https://caml.inria.fr/mantis/view.php?id=7538) -- did anyone actually compare the binary output for a given library?  does cross-library inlining work as expected/previously?

Reply:
+1 on keeping jbuilder off the list... Currently debugging this issue:
```ocaml
File "_none_", line 1:
Warning 11: this match case is unused.
File "_none_", line 1:
Warning 11: this match case is unused.
File "_none_", line 1:
Warning 11: this match case is unused.
File "lib/constants.ml", line 1:
Error: Some fatal warnings were triggered (3 occurrences)
pkg.ml: [ERROR] cmd ['jbuilder' 'build' '--root' '.' '--dev']: exited with 1
```